fred2/orienteditor.cpp

```diff
@@ -182,9 +182,27 @@ BOOL orient_editor::OnInitDialog()
 	return TRUE;
 }
 
-bool orient_editor::close(float val, const CString &str)
+bool orient_editor::close(float val, const CString &input_str)
 {
-	float diff = val - convert(str);
+	CString str;
+	str.Format("%.01f", val);
+	val = convert(str);
+
+	float input_val = convert(input_str);
```

Comment:
Is it a problem that we don't know the precision of the value encoded in `input_str`, given that we're enforcing a precision on `val` via `str`? Or do we know?

Reply:
The problem is that we do know the precision, but due to a combination of format rounding and floating point rounding, the values can drift apart.  Hopefully the new commit clarifies this.